python: Simplify black config

ddn0 opened the pull request:
Black knows to read pyproject.toml for project-wide configuration. Set
the formatting configuration there to help people with IDEs that pick up
on this or if someone wants to run black directly. Leave the command
line argument configuration around as well because some uses of
check_python_format occur in repos without pyproject.toml.

Also, remove redundant ignores and pruning. Black already ignores
directories named .git or build.

l-hoang commented on libquery/include/galois/graphs/QueryGraph.h:
```diff
@@ -20,30 +20,75 @@
+};
+
+//! Data node data type.
+struct DataNode {
```
I would move this (and all other DataNode related things) into a separate file.

roshandathathri replied:
They are related. I would prefer to keep them in the same file and change the name of the header. We can do that later.

l-hoang replied:
header file name change works for me

l-hoang commented on libquery/src/AttributedGraph.cpp:
```diff
@@ -757,15 +757,46 @@ size_t AttributedGraph::matchQuery(
   }
 
-  queryGraph.constructAndSortIndex();
+  queryGraph.sortAllEdgesByDst();
```
Why not define construct/sortIndex for the new query graph type too?

roshandathathri replied:
"index" has no specific meaning for LC_CSR_CSC_Graph and the code below the 3 CSR_CSC functions called does not fit within the graph class because it pertains to node data.

ddn0 committed:
python: Simplify black config

Black knows to read pyproject.toml for project-wide configuration. Set
the formatting configuration there to help people with IDEs that pick up
on this or if someone wants to run black directly. Leave the command
line argument configuration around as well because some uses of
check_python_format occur in repos without pyproject.toml.

Also, remove redundant ignores and pruning. Black already ignores
directories named .git or build.

arthurp commented on scripts/check_python_format.sh:
```diff
@@ -15,16 +15,9 @@ if [ "$1" == "-fix" ]; then
 fi
```
Remove `--line-length 120` on line 3.

ddn0 replied:
I can't yet because the other repo doesn't have a `pyproject.toml` file so we'd get some divergence in formatting style. But yes, soon.

arthurp reviewed (approved):
Great! My research at the time concluded that they had specifically set it up so that there were not config files (I presumed because they didn't want users to config things). I'm glad I was wrong.
